src/main/java/com/food/samosa/entity/FoodPreferenceEntity.java

```diff
@@ -0,0 +1,64 @@
+package com.food.samosa.entity;
+
+import javax.persistence.*;
+
+@Entity
+@Table(name ="food_preference")
+public class FoodPreferenceEntity extends BaseEntity {
+    @Column(length = 50)
+    private String userID;
```

Comment:
Change all these fields to be correct camel case, so userId, foodId.

Table names should also just be snake_case, without capitalization after the underscores. All lowercase.